Library/Homebrew/exceptions.rb

```diff
@@ -131,6 +148,18 @@ def initialize(name)
   end
 end
 
+class TapAlreadyUnshallowError < RuntimeError
+  attr_reader :name
+
+  def initialize(name)
+    @name = name
+
+    super <<-EOS.undent
+      Tap #{name} already full clone.
```

Comment:
We don't use "is" in the existing already-tapped or already-installed messages. This is consistent.

```
$ brew tap homebrew/dupes
Warning: Tap homebrew/dupes already tapped.
$ brew install gnu-time
Warning: gnu-time-1.7 already installed
```